contracts/CompoundFCM.sol

```diff
@@ -0,0 +1,337 @@
+  /// @dev if settlement cashflow of the trader is negative, we need to update trader's margin in terms of scaled yield bearing tokens to account the settlement casflow
+  /// @dev once settlement cashflows are accounted for, we safeTransfer the scaled yield bearing tokens in the margin account of the trader back to their wallet address
+  function settleTrader() external override onlyAfterMaturity returns (int256 traderSettlementCashflow) {
+
+    // todo: recipient as input to the function
+
+    TraderWithYieldBearingAssets.Info storage trader = traders[msg.sender];
+
+    int256 settlementCashflow = FixedAndVariableMath.calculateSettlementCashflow(trader.fixedTokenBalance, trader.variableTokenBalance, _marginEngine.termStartTimestampWad(), _marginEngine.termEndTimestampWad(), _rateOracle.variableFactor(_marginEngine.termStartTimestampWad(), _marginEngine.termEndTimestampWad()));
+    trader.updateBalancesViaDeltas(-trader.fixedTokenBalance, -trader.variableTokenBalance);
+
+    if (settlementCashflow < 0) {
+      uint256 currentExchangeRate = _ctoken.exchangeRateStored();
+      uint256 updatedTraderMarginInScaledYieldBearingTokens = trader.marginInScaledYieldBearingTokens - uint256(-settlementCashflow).wadDiv(currentExchangeRate);
+      trader.updateMarginInScaledYieldBearingTokens(updatedTraderMarginInScaledYieldBearingTokens);
+    }
+
+    // if settlement happens late, additional variable yield beyond maturity will accrue to the trader
+    uint256 traderMarginInYieldBearingTokens = getTraderMarginInYieldBearingTokens(trader.marginInScaledYieldBearingTokens);
+    trader.updateMarginInScaledYieldBearingTokens(0);
+    trader.settleTrader();
+    IERC20Minimal(address(_ctoken)).safeTransfer(msg.sender, traderMarginInYieldBearingTokens);
+    if (settlementCashflow > 0) {
+      // transfers margin in terms of underlying tokens (e.g. USDC) from the margin engine to the msg.sender
+      // as long as the margin engine is active and solvent it shoudl be able to cover the settlement cashflows of the fully collateralised traders
+      _marginEngine.transferMarginToFCMTrader(msg.sender, uint256(settlementCashflow));
+    }
+
+    return settlementCashflow;
+  }
+
+
+  /// @notice Transfer Margin (in underlying tokens) from the FCM to a MarginEngine trader
+  /// @dev in case of Compound this is done by redeeming the underlying token directly from the cToken: https://compound.finance/docs/ctokens#redeem-underlying
+  function transferMarginToMarginEngineTrader(address account, uint256 marginDeltaInUnderlyingTokens) external onlyMarginEngine whenNotPaused override {
+    if (underlyingToken.balanceOf(address(_ctoken)) >= marginDeltaInUnderlyingTokens) {
+      _ctoken.redeemUnderlying(marginDeltaInUnderlyingTokens);
+    } else {
+      IERC20Minimal(address(_ctoken)).safeTransfer(account, marginDeltaInUnderlyingTokens);
```

Comment:
does this transfer the correct number of cTokens to the msg.sender, I think the argument in here should be marginDeltaInUnderlyingTokens / cTokenExchangeRate?